qiskit_experiments/framework/composite/composite_experiment.py

```diff
@@ -40,10 +44,98 @@ def __init__(self, experiments, qubits, experiment_type=None):
+
+        Note:
+            This is transpile method for the composite experiment subclass.
+            This internally calls the transpile method of the nested experiments and
+            flattens the list of sub circuits generated by each experiment.
+            Note that transpile is called for individual circuit, and thus transpile
+            configurations and hook methods are separately applied.
+
+            No transpile configuration assumed for composite experiment object itself.
+        """
+        # Transpile each sub experiment circuit
+        circuits = list(map(lambda expr: expr.run_transpile(backend), self._experiments))
+
+        # This is not identical to the `num_qubits` when the backend is AerSimulator.
+        # In this case, usually a circuit qubit number is determined by the maximum qubit index.
+        n_qubits = max(max(sub_circ.num_qubits for sub_circ in sub_circs) for sub_circs in circuits)
+
+        # merge circuits
+        return self._flatten_circuits(circuits, n_qubits)
+
+    def run_analysis(
+        self, experiment_data: ExperimentData, job: BaseJob = None, **options
+    ) -> ExperimentData:
+        """Run analysis and update ExperimentData with analysis result.
+
+        Args:
+            experiment_data: The experiment data to analyze.
+            job: The future object of experiment result which is currently running on the backend.
+            options: Additional analysis options. Any values set here will
+                override the value from :meth:`analysis_options` for the current run.
+
+        Returns:
+            An experiment data object containing the analysis results and figures.
+
+        Raises:
+            QiskitError: When the experiment data format is not for the composite experiment.
+
+        Note:
+            This is analysis method for the composite experiment subclass.
+            This internally calls the analysis method of the nested experiments and
+            outputs a representative data entry for the composite analysis.
+            Note that analysis is called for individual experiment data, and thus analysis
+            configurations and hook methods are separately applied.
+
+            No analysis configuration assumed for composite experiment object itself.
+        """
+        if not isinstance(experiment_data, CompositeExperimentData):
+            raise QiskitError("CompositeAnalysis must be run on CompositeExperimentData.")
+
+        if len(options) > 0:
+            warnings.warn(
+                f"Analysis options for the composite experiment are provided: {options}. "
+                "Note that the provided options will override every analysis of an experiment"
+                "associated with this composite experiment.",
+                UserWarning,
+            )
+
+        return super().run_analysis(experiment_data, job, **options)
+
     @abstractmethod
-    def circuits(self, backend=None):
+    def _flatten_circuits(
```

Comment:
This method flattens for batch and merges for parallel; its name is "flatten" and its doc string says "merge". Rename to something that captures both.

Reply:
Done in https://github.com/Qiskit/qiskit-experiments/pull/380/commits/3fa62e1171f1e56e9217c4baa19d170f695af4d4 I unified to `flatten` since this term is generally used to convert a nested list to a single list.